### ISSUE TITLE
Print warnings to stderr, not stdout

### ISSUE THREAD

**Issue by hdgarrood** [spacchetti/spago#256]
At the moment I'm using `spago sources` to build a list of source globs which I then pass to `purs` directly. This mostly works great, except for the fact that some warnings get printed to stdout, which confuses things:

```
$ spago sources 2>/dev/null
WARNING: wasn't able to check if your package set file is frozen
.spago/aff/v5.1.1/src/**/*.purs
.spago/aff-coroutines/v7.0.0/src/**/*.purs
.spago/arrays/v5.3.0/src/**/*.purs
.spago/avar/v3.0.0/src/**/*.purs
.spago/bifunctors/v4.0.0/src/**/*.purs
```

This leads to a bunch of warnings such as:

```
> set -o noglob && purs compile $(npm run --silent sources:purs | tr '\n' ' ')

purs compile: No files found using pattern: WARNING:
purs compile: No files found using pattern: wasn't
purs compile: No files found using pattern: able
purs compile: No files found using pattern: to
purs compile: No files found using pattern: check
purs compile: No files found using pattern: if
purs compile: No files found using pattern: your
purs compile: No files found using pattern: package
purs compile: No files found using pattern: set
purs compile: No files found using pattern: file
purs compile: No files found using pattern: is
purs compile: No files found using pattern: frozen
```

Incidentally, once #173 is a thing I can probably just use that and not need to worry about this.

**Comment by f-f** [comment]
Good point. I think that since in some cases the output on `stdout` should be meaningful for machines too then the program should log _everything_ on `stderr`.

I think this is also the easiest thing to do, since it's just about having [`echo`](https://github.com/spacchetti/spago/blob/d5b6d4620b56c2dd8b64fa635eb4b340815ddfda/src/Spago/Prelude.hs#L128) print to `stderr`

**Comment by hdgarrood** [comment]
I've realised one potential issue with doing this, which is that the compiler currently prints json errors to stderr, so if you run `spago build -- --json-errors` then you'll get both the json errors and spago logging sent to the same stream.

I suppose one option is to also provide a `--silent` flag, like `npm run --silent`, which suppresses all output from spago itself so that you only get output from subcommands. I guess the compiler ought to print json errors to stdout really too, but that might be a bit awkward to change at this point.

**Comment by hdgarrood** [comment]
See also https://github.com/purescript/purescript/issues/3672

**Comment by f-f** [comment]
Update: as we figured in #316, there is also the output from external commands, e.g. `purs`, getting in the way sometimes. So I think we should just output _all_ logging to `stderr` (including piping output from external commands) except when printing actual "business" output (e.g. in case of `spago sources`, `spago list-packages`, `spago run`, etc)

I think we should just remove the current print facilities (`echo` and `echoDebug`) and replace them with `output` (that goes to `stdout`) and various "log" functions going to `stderr`: `log`, `logDebug`, `logWarning`, `logError` (these last two are the same as `log` except they prepend "WARNING: " etc to the log message. Eventually we could switch to some proper logging thing, e.g. I was considering [`RIO`](https://haskell.fpcomplete.com/library/rio#logging))

**Comment by Benjmhart** [comment]
I'll try and tackle this in the next week or two

**Comment by f-f** [comment]
Fixed in #475 and #486 🎉
